### ISSUE TITLE
Feature request: Allow getting URLs to the 'blame' version of lines of code 

### ISSUE THREAD

**Issue by zlamma** [reduckted/GitWebLinksVS2019#3]
Hi. This is elaborating on [my suggestion from the Q&A page](https://marketplace.visualstudio.com/items?itemName=reduckted.GitWebLinks#qna), which you have amazingly implemented so quickly 👍.

I know I am being an ungrateful bastard, but I just wanted to leave for your consideration this extra improvement.

Namely, in the original suggestion I was justifying that the reason I would prefer SHAs is because I would like something that will work for as long as possible (so, close to a permalink).   

But, for the likes of me, who don't shy away from rewriting my commits to try to achieve the most readable pull request and the most helpful history, using the SHA of last commit is still going to have a problem. That is, when I reorganize the history of my branch, the SHAs will be removed from Github and links will stop working. Same problem will also happen to people who use the coarse-grained history edits during a pull (provided both by [GitHub](https://github.com/blog/2243-rebase-and-merge-pull-requests) and [Bitbucket](https://blog.bitbucket.org/2017/01/31/git-squash-commits-merging-bitbucket/)).

Hence my suggestion was to allow to use the SHA from the 'blame' command of the line of code (or Annotate in VisualStudio's lingo). Effectively, that will provide 'the oldest SHA that had this line', which should guarantee the most stable URL that is available.

Now, I know this might not be something that everyone would want all the time, especially if they are interested in discussing more than one line, each of which can come from a different commit, so perhaps both options could be provided.

I know it might not be trivial to implement this, but perhaps it's not that hard either. Maybe VS's Annotate or commandline's "git blame" already offer the underlying 'meat'?

**Comment by reduckted** [comment]
I'm not 100% sure I understand what you're after. Would you be able to give me an example of the URL that you would want this to produce?

If we take this file, as an example:
https://github.com/reduckted/GitWebLinks/blob/master/source/GitWebLinks/Links/Handlers/GitHubHandler.vb#L10

The extension can create the link to a specific commit:
`https://github.com/reduckted/GitWebLinks/blob/8764e0f3f3cba872835b43a782e8493a599ae054/source/GitWebLinks/Links/Handlers/GitHubHandler.vb#10`
(I've escaped it because GitHub now renders the code instead of the URL 😮) 

Here's the blame URL for the file:
https://github.com/reduckted/GitWebLinks/blame/master/source/GitWebLinks/Links/Handlers/GitHubHandler.vb#L10

Are you wanting the blame URL for a specific commit?
https://github.com/reduckted/GitWebLinks/blame/8764e0f3f3cba872835b43a782e8493a599ae054/source/GitWebLinks/Links/Handlers/GitHubHandler.vb#L10

**Comment by zlamma** [comment]
Ah! The confusion is about the blame! Thanks for inquiring!

This is not a request for links to blame-pages, but rather to the same URLs as the original functionality, just using different SHAs. Namely, for right click on line of code, it would be an extra option akin to 'Copy link to this line, at file version where the line was introduced'.
For your file, if I used it on [this line](https://github.com/reduckted/GitWebLinks/blob/04db5de/source/GitWebLinks/Links/Handlers/GitHubHandler.vb#L40), it should produce [this link](https://github.com/reduckted/GitWebLinks/blob/b0026ec6b03360b45e7cce13d9368b7e34ac4b1f/source/GitWebLinks/Links/Handlers/GitHubHandler.vb#L30), which essentially uses the SHA of the commit 'Used options for the server URLs.', as visible in [this blame of that line](https://github.com/reduckted/GitWebLinks/blame/04db5de/source/GitWebLinks/Links/Handlers/GitHubHandler.vb#L40), (equivalent of command `git blame source/GitWebLinks/Links/Handlers/GitHubHandler.vb`).

A similar option could be introduced for the whole file, where it would use the SHA of the latest commit that actually modified that file, rather than just any latest commit.

I find such links most useful, because both 'latest commit SHA' and 'branch' often end up getting wiped out from the history (examples in [my original comment](#issue-276197921) near _'But, for the likes of me...'_).

**Comment by reduckted** [comment]
Ah, OK, I think I understand what you're after now. This would only really work well if you haven't changed the code that you're creating the link to. If you have changed that code, then the SHA would _still_ change when you rewrite the history. Wouldn't that be a problem?

**Comment by zlamma** [comment]
Yes indeed. They would be just 'the most stable URLs available'. In cases of linking to lines only just added by your own private commit, you would end up surprised. But this is rather just an absent-minded thing to do, not the common use. Currently you end up surprised every time you even have any private commit in your repo.

**Comment by slawomir-brzezinski-at-interxion** [comment]
After a thought, I feel that getting the SHA of the last commit on the given file would deliver most of the value, so can be considered a first quick win, in that this would no longer be the case:

_"@zlamma: Currently you end up surprised every time you even have any private commit in your repo."_
